entry_types/scrolled/package/src/frontend/inlineEditing/PhonePlatformProvider.js

```diff
@@ -0,0 +1,30 @@
+import React, {useEffect, useState} from 'react';
+
+import PhonePlatformContext from '../PhonePlatformContext';
+
+export function PhonePlatformProvider({children}) {
+
+  const [phoneEmulationMode, setPhoneEmulationMode] = useState(false);
+
+  useEffect(() => {
+    window.addEventListener('message', receive);
+
+    function receive(event) {
+      if (event.data.type === 'CHANGE_EMULATION_MODE') {
+        if(event.data['payload'] === 'phone') {
+          setPhoneEmulationMode(true);
+        }
+        else {
+          setPhoneEmulationMode(false);
+        }
+      }
+    }
+    return () => document.removeEventListener('message', receive);
```

Comment:
**issue**: inconsistent event registration

Event is added to `window` and removed from `document`. If we had a spec that unmounts the component and post a message afterwards, this would have lead to a warning about manipulating state on an unmounted component.